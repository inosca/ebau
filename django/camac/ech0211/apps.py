from django.apps import AppConfig


class Ech0211Config(AppConfig):
    name = "camac.ech0211"

    def ready(self):
        import camac.patches  # noqa
        import camac.ech0211.event_handlers  # noqa
