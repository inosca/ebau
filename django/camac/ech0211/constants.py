# eCH0210 - 3.4 Entscheid / Verf√ºgung - decisionRuling
## 2.4.1.1 Beurteilung - judgement
ECH_JUDGEMENT_APPROVED = 1  # positiver Entscheid
ECH_JUDGEMENT_APPROVED_WITH_RESERVATION = 2  # positiv mit Bedinungen/Auflagen
ECH_JUDGEMENT_WRITTEN_OFF = 3  # Nicht-Eintreten
ECH_JUDGEMENT_DECLINED = 4  # abgelehnt/negativer Enscheid


ECH_API_LEVEL_FULL = "full"
ECH_API_LEVEL_BASIC = "basic"
