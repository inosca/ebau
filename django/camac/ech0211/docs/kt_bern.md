# Gemeindeschnittstelle eBau

## Authentifizierung

Zur Authentifizierung wird der Standard [OpenID Connect](https://openid.net/connect/) genutzt. Pro Gemeinde wird eine `client-id` und ein `client-secret` vergeben, mit welchen Tokens bezogen werden können:

```bash
curl --request POST \
--url 'https://portal.ebau-test.sycloud.ch/auth/realms/camac/protocol/openid-connect/token' \
--header 'content-type: application/x-www-form-urlencoded' \
--data grant_type=client_credentials \
--data scope=openid \
--data client_id='${client-id}' \
--data client_secret=${client-secret}
```

Mit einem gültigen Token können API-Abfragen gemacht werden. Nachfolgend ein paar Beispiele:

## Sichtbare Gesuche auflisten

```bash
curl -X GET "https://ebau-test.sycloud.ch/ech/v1/applications?group=123" -H "Authorization: Bearer ${TOKEN}"
```

## Gesuch-Details abfragen (BaseDelivery)

```bash
curl -X GET "https://ebau-test.sycloud.ch/ech/v1/application/XYZ?group=123" -H "Authorization: Bearer ${TOKEN}"
```

## Group Parameter

Bei den meisten Endpunkten, lässt sich über den `group` Parameter angeben, im Namen welcher Gruppe ein Request gemacht werden soll.

Dies ist notwendig, um die verschiedenen Stellen innerhalb einer Gemeinde zu unterscheiden. Alle Meldungen nach Erteilung eines positiven Bauentscheids werden an die Baukontrolle adressiert.

### Beispiel

- Gemeinde client `A` hat als Standardgruppe `Leitung Leitbehörde Gemeinde A`
- Zusätzlich besteht noch eine Mitgliedschaft in der Gruppe `Leitung Baukontrolle Gemeinde A`
- Requests mit dem client dieser Gemeinde werden standardmässig im Namen der Gruppe `Leitung Leitbehörde Gemeinde A` gemacht
- Sollen nun beispielsweise Meldungen für die Gruppe `Leitung Baukontrolle Gemeinde A` abgeholt werden, muss der `group` Parameter entsprechend gesetzt werden

### Abfragen von Gruppen IDs

Unter dem Tag [User](#/User) sind die Endpunkte zusammengefasst, die eine Abfrage von Gruppenmitgliedschaften und -IDs ermöglichen:

- [/me](#/User/api_v1_me_read) zeigt Informationen zum aktuellen User, inklusive Gruppenmitgliedschaften, an
- [/groups](#/User/api_v1_groups_list) zeigt alle Gruppen an, für welche eine Mitgliedschaft besteht
- [/groups/{group_id}](#/User/api_v1_groups_read) zeigt Informationen zu einer spezifischen Gruppe an

### Direktlinks in eBau

Verschiedene Aufgaben werden gemäss Spezifikation direkt in eBau erledigt. Unterstützt werden folgende Links:

- `/ech/v1/instance/<instance_id>/`: Allgemeiner Link auf ein Dossier, Einstieg für Dossierkorrektur (Kap. 3.1)
- `/ech/v1/ebau-number/<instance_id>/`: eBau-Nummer vergeben (Kap. 3.1)
- `/ech/v1/claim/<instance_id>/`: Nachforderungsseite (Kap. 3.1, 3.3.2)
- `/ech/v1/dossier-check/<instance_id>/`: Dossierprüfung (formelle und materielle Prüfung, Kap. 3.1)
- `/ech/v1/revision-history/<instance_id>/`: Änderungsverlauf von Dossier

## Abweichungen und Besonderheiten

- Der Type `planningPermissionAplicationIdentifier` enthält unter `localId` die eBau-Nummer und unter `dossierIdentification` unsere Dossiernummer ("Instance id")

- Der Type `localOrganisationId` enthält unter `organisationId` unsere Service id. Service ids können über den `/ech/v1/public-services/` endpoint abgefragt werden.

- Der von uns empfangene Typ `task`, um Stellungnahmen anzufordern (Spezifikation 3.2) muss zwingend die Service id der einzuladenden Stelle im `extension` Typ enthalten:

  ```xml
  <ns2:extension>
    <serviceId>23</serviceId>
  </ns2:extension>
  ```

- Der eCH-Standard forciert, dass bei den meisten Meldungen ein `document` mitgeschickt wird. Dieses `document` wird (mit Ausnahme von `accompanyingReport`) von eBau ignoriert. Dokumente werden über unsere API hoch- und heruntergeladen. Beim Hochladen werden sie bereits einer `Instance`, sowie einer oder mehreren `AttachmentSection` zugewiesen. Somit sind Dokumente in eCH Meldungen, die von eBau erhalten werden, redundant und werden ignoriert.

  Bei ausgehenden Meldungen werden die Dokumente jedoch korrekt abgefüllt. Dabei gilt zu beachten:

  - `documentKind` enthält alle `AttachmentSections`, separiert durch `; `
  - `keywords` enthält alle Tags (zB: `vollmacht-dokument`)
  - `uuid` enthält eine mit dem `Attachment` assoziierte uuid, diese ist jedoch ansonstenn nicht über die API exposed. `Attachments` werden prinzipiell über ihren PK referenziert

- In einer `application` wird immer der Status `6701` gesetzt. Der korrekte Status aus dem eBau findet sich unter `namedMetaData.status`. Bei einer `statusNotification` wird immer der Status `in progress` gesetzt. Der korrekte Status findet sich im `remark`.

- `buildingCategory` wird immer auf `1040` gesetzt.

- `documentStatusType` wird immer auf `signed` gesetzt.

- `realestateType` wird immer auf `8` gesetzt.

- Properties, die in eCH zwingend sind, in eBau jedoch nicht, werden bei nichtvorhandensein mit `unknown` befüllt.

- In eBau ist es möglich, auch ausländische Adressen zu erfassen. Bei solchen wird die Postleitzahl auf `9999` gesetzt, falls sie nicht vierstellig sein sollte.

- In eCH ist es nicht möglich, Kosten von unter 1000.- zu erfassen. Sollten die in eBau erfassten Kosten tiefer als 1000.- sein, wird `1000` eingetragen.

- Relevante URLs finden sich in den Meldungen im `HeaderType` unter `extension`.

- 4.2 Bauverfahren abschliessen: Bei Voranfragen wird der Prozess mit einem Entscheid (bzw. fachlich einer Beurteilung) abgeschlossen. "close dossier" hat entgegen der Spezifikation für Voranfragen also keine Bedeutung. Stattdessen wird der Prozess mit "notice ruling" abgeschlossen (siehe Kap. 3.2).

- 5.3.5 Rückzug des Baugesuchs: Diese Funktion ist nicht in eBau implementiert, darum kann diese Message nicht ausgeliefert werden.

- Judgements in `NoticeRuling` werden in eBau wie folgt gemappt:

  | Judgement | Descision               | Besonderheiten                     |
  | --------- | ----------------------- | ---------------------------------- |
  | 1         | Bewilligt               |                                    |
  | 2         | Bewilligt mit Vorbehalt | Nicht verfügbar für Baugesuche     |
  | 3         | Abgeschrieben           | Nicht verfügbar für Vorabklärungen |
  | 4         | Abgelehnt               |                                    |

- `relationshipToPerson`: Hier füllen wir folgende Personalien ab:

  | eBau form               | ech role       |
  | ----------------------- | -------------- |
  | Gesuchsteller           | applicant      |
  | Vertreter mit Vollmacht | contact        |
  | Projektverfasser        | project author |
  | Grundeigentümer         | landowner      |

  Sollte es sich bei einem Eintrag um eine juristische Person handeln, sieht eCH keine
  Felder vor für Vor- und Nachname der Kontaktperson. Wir füllen diese Namen daher in das Feld
  `organisationAdditionalName` im Format: "Vorname Name".

## Message Typen

### GET

{get_messages}

### POST

Der Parameter `messageType` wird bei der Auswertung ignoriert - entscheidend ist der `event type` sowie, falls nötig, zusätzliche Parameter (z.B. `judgement` bei `NoticeRuling`).

{post_messages}
