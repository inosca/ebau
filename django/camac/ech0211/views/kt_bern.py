import logging

from django.http import HttpResponse
from drf_yasg import openapi
from drf_yasg.utils import swagger_auto_schema
from pyxb import IncompleteElementContentError, UnprocessedElementContentError
from rest_framework.authentication import get_authorization_header
from rest_framework.generics import get_object_or_404
from rest_framework.mixins import ListModelMixin, RetrieveModelMixin
from rest_framework.parsers import JSONParser
from rest_framework.serializers import Serializer
from rest_framework.viewsets import GenericViewSet
from rest_framework_xml.renderers import XMLRenderer

from camac.constants.kt_bern import ECH_BASE_DELIVERY
from camac.instance.models import Instance
from camac.swagger.utils import get_operation_description, group_param

from .. import event_handlers, formatters
from ..data_preparation import get_document
from ..mixins import ECHInstanceQuerysetMixin
from ..parsers import ECHXMLParser
from ..send_handlers import SendHandlerException, get_send_handler
from ..serializers import ApplicationsSerializer

logger = logging.getLogger(__name__)


class ApplicationView(ECHInstanceQuerysetMixin, RetrieveModelMixin, GenericViewSet):
    instance_field = None
    serializer_class = Serializer
    renderer_classes = (XMLRenderer,)
    queryset = Instance.objects
    instance_field = None

    @swagger_auto_schema(
        tags=["ECH"],
        manual_parameters=[group_param],
        operation_summary="Get baseDelivery for instance",
        operation_description=get_operation_description(),
        responses={"200": "eCH-0211 baseDelivery"},
    )
    def retrieve(self, request, instance_id=None, **kwargs):
        qs = self.get_queryset()
        instance = get_object_or_404(qs, pk=instance_id)
        document = get_document(instance.pk)
        base_delivery_formatter = formatters.BaseDeliveryFormatter(config="kt_bern")
        try:
            xml_data = formatters.delivery(
                instance,
                document,
                message_type=ECH_BASE_DELIVERY,
                eventBaseDelivery=base_delivery_formatter.format_base_delivery(
                    instance, answers=document
                ),
            ).toxml()
        except (
            IncompleteElementContentError,
            UnprocessedElementContentError,
        ) as e:  # pragma: no cover
            logger.error(e.details())
            raise
        response = HttpResponse(xml_data)
        response["Content-Type"] = "application/xml"

        return response


class ApplicationsView(ECHInstanceQuerysetMixin, ListModelMixin, GenericViewSet):
    instance_field = None
    serializer_class = ApplicationsSerializer
    queryset = Instance.objects
    instance_field = None
    filter_backends = []

    def get_queryset(self, group=None):
        if getattr(self, "swagger_fake_view", False):  # pragma: no cover
            return Instance.objects.none()
        return super().get_queryset()

    @swagger_auto_schema(
        tags=["ECH"],
        manual_parameters=[group_param],
        operation_summary="Get list of accessible instances",
        operation_description=get_operation_description(),
    )
    def list(self, request, *args, **kwargs):
        return super().list(request, *args, **kwargs)


last_param = openapi.Parameter(
    "last",
    openapi.IN_QUERY,
    description=(
        "UUID of last message. Can be found in `delivery.deliveryHeader.messageId`. "
        "If omitted, first message is returned"
    ),
    type=openapi.TYPE_STRING,
)


class EventView(ECHInstanceQuerysetMixin, GenericViewSet):
    instance_field = None
    queryset = Instance.objects
    swagger_schema = None
    parser_classes = (JSONParser,)
    serializer_class = Serializer

    def has_create_permission(self):
        if self.request.group.role.name == "support":
            return True
        return False

    def create(self, request, instance_id, event_type, *args, **kwargs):
        instance = get_object_or_404(self.get_queryset(), pk=instance_id)
        try:
            EventHandler = getattr(event_handlers, f"{event_type}EventHandler")
        except AttributeError:
            return HttpResponse(status=404)
        eh = EventHandler(
            instance=instance,
            user_pk=request.user.pk,
            group_pk=request.group.pk,
        )
        try:
            eh.run()
        except event_handlers.EventHandlerException as e:
            return HttpResponse(str(e), status=400)
        return HttpResponse(status=201)


class SendView(ECHInstanceQuerysetMixin, GenericViewSet):
    instance_field = None
    queryset = Instance.objects
    renderer_classes = (XMLRenderer,)
    parser_classes = (ECHXMLParser,)
    serializer_class = Serializer

    @swagger_auto_schema(
        tags=["ECH"],
        manual_parameters=[group_param],
        operation_summary="Send message",
        operation_description=get_operation_description(),
        request_body=openapi.Schema(
            type=openapi.TYPE_STRING,
            description="An event wrapped in a [eCH-0211-Delivery](https://www.ech.ch/standards/60526).",
        ),
        responses={"201": "success"},
    )
    def create(self, request, *args, **kwargs):
        if not request.data:
            return HttpResponse(status=400)

        try:
            send_handler = get_send_handler(
                request.data,
                self.get_queryset(),
                request.user,
                request.group,
                get_authorization_header(request),
                request.caluma_info.context.user,
            )
        except SendHandlerException as e:
            return HttpResponse(str(e), status=404)

        has_perm, msg = send_handler.has_permission()

        if not has_perm:
            return HttpResponse(msg, status=403)

        try:
            send_handler.apply()
        except SendHandlerException as e:
            return HttpResponse(str(e), status=e.status)

        return HttpResponse(status=201)
