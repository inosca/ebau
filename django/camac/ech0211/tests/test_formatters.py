import logging
import os.path

import pytest
import xmlschema
from caluma.caluma_workflow.api import (
    cancel_work_item,
    complete_work_item,
    skip_work_item,
)
from django.core.management import call_command
from pyxb import IncompleteElementContentError, UnprocessedElementContentError

from camac.constants.kt_bern import ECH_BASE_DELIVERY
from camac.document.models import Attachment
from camac.ech0211 import formatters
from camac.ech0211.formatters import determine_decision_state

logger = logging.getLogger(__name__)


@pytest.mark.parametrize(
    "form",
    ["baugesuch", "einfache vorabklaerung", "vollstaendige vorabklaerung"],
)
def test_base_delivery(
    form,
    ech_mandatory_answers_baugesuch,
    ech_mandatory_answers_einfache_vorabklaerung,
    ech_mandatory_answers_vollstaendige_vorabklaerung,
    set_application_be,
    ech_instance_be,
    multilang,
):

    ech_mandatory_answers = ech_mandatory_answers_baugesuch
    if form == "baugesuch":
        ech_mandatory_answers_baugesuch["baukosten-in-chf"] = 999  # too cheap
        ech_mandatory_answers_baugesuch["parzelle"][0][
            "lagekoordinaten-nord"
        ] = 1070500.0001  # too many decimal places
        ech_mandatory_answers_baugesuch["parzelle"][0][
            "lagekoordinaten-ost"
        ] = 2480034.0
    elif form == "einfache vorabklaerung":
        ech_mandatory_answers = ech_mandatory_answers_einfache_vorabklaerung
        ech_mandatory_answers[
            "ort-grundstueck"
        ] = ""  # implicitly test filling up strings to min_length
        # implicitly test empty coordinates
        ech_mandatory_answers["parzelle"][0]["lagekoordinaten-nord"] = None
        ech_mandatory_answers["parzelle"][0]["lagekoordinaten-ost"] = None

    elif form == "vollstaendige vorabklaerung":
        ech_mandatory_answers = ech_mandatory_answers_vollstaendige_vorabklaerung

    configured_base_delivery_formatter = formatters.BaseDeliveryFormatter(
        config="kt_bern"
    )

    xml = formatters.delivery(
        ech_instance_be,
        ech_mandatory_answers,
        ECH_BASE_DELIVERY,
        eventBaseDelivery=configured_base_delivery_formatter.format_base_delivery(
            ech_instance_be, answers=ech_mandatory_answers
        ),
    )

    assert xml

    my_dir = os.path.dirname(__file__)
    my_schema = xmlschema.XMLSchema(my_dir + "/../xsd/ech_0211_2_0.xsd")
    try:
        xml_data = xml.toxml()
    except (
        IncompleteElementContentError,
        UnprocessedElementContentError,
    ) as e:  # pragma: no cover
        logger.error(e.details())
        raise

    my_schema.validate(xml_data)


def test_office(ech_instance_be, ech_snapshot, multilang):
    off = formatters.office(
        ech_instance_be.responsible_service(filter_type="municipality"),
        organization_category="ebaube",
        canton="BE",
    )
    ech_snapshot(off.toxml(element_name="office"))


@pytest.mark.parametrize("amount", [0, 1, 2])
@pytest.mark.parametrize("with_display_name", [True, False])
def test_get_documents(db, attachment_factory, amount, with_display_name, ech_snapshot):
    context = {}
    if with_display_name:
        context = {"displayName": "baz"}
    uuids = [
        "7604864d-fada-4431-b63b-fc9f4915233d",
        "23daf554-c2f5-4aa2-b5f2-734a96ed84d8",
    ]
    for count in range(1, amount + 1):
        attachment_factory(
            name="foo.bar",
            context=context,
            attachment_id=count,
            uuid=uuids[count - 1],
            mime_type="application/pdf",
        )

    xml = formatters.get_documents(Attachment.objects.filter(uuid__in=uuids))

    assert xml

    for doc in xml:
        try:
            ech_snapshot(doc.toxml(element_name="doc"))
        except (
            IncompleteElementContentError,
            UnprocessedElementContentError,
        ) as e:  # pragma: no cover
            logger.error(e.details())
            raise


@pytest.mark.freeze_time("2022-01-01")
@pytest.mark.parametrize(
    "skip,task_id,work_item_action,expected_decision,expected_state",
    [
        (
            ["submit", "complete-check", "skip-circulation"],
            "make-decision",
            cancel_work_item,
            4,  # negative
            "denied",
        ),
        (
            ["submit", "complete-check", "skip-circulation"],
            "make-decision",
            complete_work_item,
            1,  # positive
            "accepted",
        ),
        (["submit"], "reject-form", complete_work_item, 3, "negative"),
    ],
)
def test_decision_formatter(
    ech_instance_sz,
    instance_state_factory,
    caluma_config_sz,
    set_application_sz,
    skip,
    task_id,
    work_item_action,
    expected_decision,
    expected_state,
    caluma_admin_user,
):
    call_command("loaddata", "/app/kt_schwyz/config/instance.json")

    instance_state_factory(name=expected_state)
    for task in skip:
        t = ech_instance_sz.case.work_items.get(task_id=task)
        skip_work_item(t, caluma_admin_user)
    work_item_action(
        ech_instance_sz.case.work_items.get(task_id=task_id),
        caluma_admin_user,
        {"no-notification": True},
    )
    decision, decision_date = determine_decision_state(ech_instance_sz)
    assert decision == expected_decision
