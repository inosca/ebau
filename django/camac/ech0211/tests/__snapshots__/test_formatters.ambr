# serializer version: 1
# name: test_get_alexandria_documents
  '''
  <?xml version="1.0" ?>
  <doc xmlns:ns1="http://www.ech.ch/xmlns/eCH-0147/T0/1" xmlns:ns2="http://www.ech.ch/xmlns/eCH-0039/2">
  	<ns1:uuid>394f53af-24bc-4324-986e-c3901c310263</ns1:uuid>
  	<ns1:titles>
  		<ns2:title>Jeffrey Santos</ns2:title>
  	</ns1:titles>
  	<ns1:status>created</ns1:status>
  	<ns1:files>
  		<ns1:file>
  			<ns1:pathFileName>http://ebau.local/ech/v1/files/<!-- ATTACHMENT_ID --></ns1:pathFileName>
  			<ns1:mimeType>application/octet-stream</ns1:mimeType>
  		</ns1:file>
  	</ns1:files>
  	<ns1:documentKind>Susan Green</ns1:documentKind>
  </doc>
  
  '''
# ---
# name: test_get_alexandria_documents.1
  '''
  <?xml version="1.0" ?>
  <doc xmlns:ns1="http://www.ech.ch/xmlns/eCH-0147/T0/1" xmlns:ns2="http://www.ech.ch/xmlns/eCH-0039/2">
  	<ns1:uuid>23daf554-c2f5-4aa2-b5f2-734a96ed84d8</ns1:uuid>
  	<ns1:titles>
  		<ns2:title>Nancy Delacruz</ns2:title>
  	</ns1:titles>
  	<ns1:status>approved</ns1:status>
  	<ns1:files>
  		<ns1:file>
  			<ns1:pathFileName>http://ebau.local/ech/v1/files/<!-- ATTACHMENT_ID --></ns1:pathFileName>
  			<ns1:mimeType>application/octet-stream</ns1:mimeType>
  		</ns1:file>
  	</ns1:files>
  	<ns1:documentKind>Michelle Allen</ns1:documentKind>
  </doc>
  
  '''
# ---
# name: test_get_alexandria_documents.2
  '''
  <?xml version="1.0" ?>
  <doc xmlns:ns1="http://www.ech.ch/xmlns/eCH-0147/T0/1" xmlns:ns2="http://www.ech.ch/xmlns/eCH-0039/2">
  	<ns1:uuid>7604864d-fada-4431-b63b-fc9f4915233d</ns1:uuid>
  	<ns1:titles>
  		<ns2:title>Aaron Davis</ns2:title>
  	</ns1:titles>
  	<ns1:status>invalidated</ns1:status>
  	<ns1:files>
  		<ns1:file>
  			<ns1:pathFileName>http://ebau.local/ech/v1/files/<!-- ATTACHMENT_ID --></ns1:pathFileName>
  			<ns1:mimeType>application/octet-stream</ns1:mimeType>
  		</ns1:file>
  	</ns1:files>
  	<ns1:documentKind>Bryan Fields</ns1:documentKind>
  </doc>
  
  '''
# ---
# name: test_get_documents[False-0]
  '''
  <?xml version="1.0" ?>
  <doc xmlns:ns1="http://www.ech.ch/xmlns/eCH-0147/T0/1" xmlns:ns2="http://www.ech.ch/xmlns/eCH-0039/2">
  	<ns1:uuid>00000000-0000-0000-0000-000000000000</ns1:uuid>
  	<ns1:titles>
  		<ns2:title>dummy</ns2:title>
  	</ns1:titles>
  	<ns1:status>signed</ns1:status>
  	<ns1:files>
  		<ns1:file>
  			<ns1:pathFileName>unknown</ns1:pathFileName>
  			<ns1:mimeType>unknown</ns1:mimeType>
  		</ns1:file>
  	</ns1:files>
  </doc>
  
  '''
# ---
# name: test_get_documents[False-1]
  '''
  <?xml version="1.0" ?>
  <doc xmlns:ns1="http://www.ech.ch/xmlns/eCH-0147/T0/1" xmlns:ns2="http://www.ech.ch/xmlns/eCH-0039/2">
  	<ns1:uuid>7604864d-fada-4431-b63b-fc9f4915233d</ns1:uuid>
  	<ns1:titles>
  		<ns2:title>foo</ns2:title>
  	</ns1:titles>
  	<ns1:status>signed</ns1:status>
  	<ns1:files>
  		<ns1:file>
  			<ns1:pathFileName>http://ebau.local/api/v1/attachments/files/?attachments=<!-- ATTACHMENT_ID --></ns1:pathFileName>
  			<ns1:mimeType>application/pdf</ns1:mimeType>
  		</ns1:file>
  	</ns1:files>
  	<ns1:documentKind/>
  </doc>
  
  '''
# ---
# name: test_get_documents[False-2]
  '''
  <?xml version="1.0" ?>
  <doc xmlns:ns1="http://www.ech.ch/xmlns/eCH-0147/T0/1" xmlns:ns2="http://www.ech.ch/xmlns/eCH-0039/2">
  	<ns1:uuid>23daf554-c2f5-4aa2-b5f2-734a96ed84d8</ns1:uuid>
  	<ns1:titles>
  		<ns2:title>foo</ns2:title>
  	</ns1:titles>
  	<ns1:status>signed</ns1:status>
  	<ns1:files>
  		<ns1:file>
  			<ns1:pathFileName>http://ebau.local/api/v1/attachments/files/?attachments=<!-- ATTACHMENT_ID --></ns1:pathFileName>
  			<ns1:mimeType>application/pdf</ns1:mimeType>
  		</ns1:file>
  	</ns1:files>
  	<ns1:documentKind/>
  </doc>
  
  '''
# ---
# name: test_get_documents[False-2].1
  '''
  <?xml version="1.0" ?>
  <doc xmlns:ns1="http://www.ech.ch/xmlns/eCH-0147/T0/1" xmlns:ns2="http://www.ech.ch/xmlns/eCH-0039/2">
  	<ns1:uuid>7604864d-fada-4431-b63b-fc9f4915233d</ns1:uuid>
  	<ns1:titles>
  		<ns2:title>foo</ns2:title>
  	</ns1:titles>
  	<ns1:status>signed</ns1:status>
  	<ns1:files>
  		<ns1:file>
  			<ns1:pathFileName>http://ebau.local/api/v1/attachments/files/?attachments=<!-- ATTACHMENT_ID --></ns1:pathFileName>
  			<ns1:mimeType>application/pdf</ns1:mimeType>
  		</ns1:file>
  	</ns1:files>
  	<ns1:documentKind/>
  </doc>
  
  '''
# ---
# name: test_get_documents[True-0]
  '''
  <?xml version="1.0" ?>
  <doc xmlns:ns1="http://www.ech.ch/xmlns/eCH-0147/T0/1" xmlns:ns2="http://www.ech.ch/xmlns/eCH-0039/2">
  	<ns1:uuid>00000000-0000-0000-0000-000000000000</ns1:uuid>
  	<ns1:titles>
  		<ns2:title>dummy</ns2:title>
  	</ns1:titles>
  	<ns1:status>signed</ns1:status>
  	<ns1:files>
  		<ns1:file>
  			<ns1:pathFileName>unknown</ns1:pathFileName>
  			<ns1:mimeType>unknown</ns1:mimeType>
  		</ns1:file>
  	</ns1:files>
  </doc>
  
  '''
# ---
# name: test_get_documents[True-1]
  '''
  <?xml version="1.0" ?>
  <doc xmlns:ns1="http://www.ech.ch/xmlns/eCH-0147/T0/1" xmlns:ns2="http://www.ech.ch/xmlns/eCH-0039/2">
  	<ns1:uuid>7604864d-fada-4431-b63b-fc9f4915233d</ns1:uuid>
  	<ns1:titles>
  		<ns2:title>baz</ns2:title>
  	</ns1:titles>
  	<ns1:status>signed</ns1:status>
  	<ns1:files>
  		<ns1:file>
  			<ns1:pathFileName>http://ebau.local/api/v1/attachments/files/?attachments=<!-- ATTACHMENT_ID --></ns1:pathFileName>
  			<ns1:mimeType>application/pdf</ns1:mimeType>
  		</ns1:file>
  	</ns1:files>
  	<ns1:documentKind/>
  </doc>
  
  '''
# ---
# name: test_get_documents[True-2]
  '''
  <?xml version="1.0" ?>
  <doc xmlns:ns1="http://www.ech.ch/xmlns/eCH-0147/T0/1" xmlns:ns2="http://www.ech.ch/xmlns/eCH-0039/2">
  	<ns1:uuid>23daf554-c2f5-4aa2-b5f2-734a96ed84d8</ns1:uuid>
  	<ns1:titles>
  		<ns2:title>baz</ns2:title>
  	</ns1:titles>
  	<ns1:status>signed</ns1:status>
  	<ns1:files>
  		<ns1:file>
  			<ns1:pathFileName>http://ebau.local/api/v1/attachments/files/?attachments=<!-- ATTACHMENT_ID --></ns1:pathFileName>
  			<ns1:mimeType>application/pdf</ns1:mimeType>
  		</ns1:file>
  	</ns1:files>
  	<ns1:documentKind/>
  </doc>
  
  '''
# ---
# name: test_get_documents[True-2].1
  '''
  <?xml version="1.0" ?>
  <doc xmlns:ns1="http://www.ech.ch/xmlns/eCH-0147/T0/1" xmlns:ns2="http://www.ech.ch/xmlns/eCH-0039/2">
  	<ns1:uuid>7604864d-fada-4431-b63b-fc9f4915233d</ns1:uuid>
  	<ns1:titles>
  		<ns2:title>baz</ns2:title>
  	</ns1:titles>
  	<ns1:status>signed</ns1:status>
  	<ns1:files>
  		<ns1:file>
  			<ns1:pathFileName>http://ebau.local/api/v1/attachments/files/?attachments=<!-- ATTACHMENT_ID --></ns1:pathFileName>
  			<ns1:mimeType>application/pdf</ns1:mimeType>
  		</ns1:file>
  	</ns1:files>
  	<ns1:documentKind/>
  </doc>
  
  '''
# ---
# name: test_office
  '''
  <?xml version="1.0" ?>
  <office xmlns:ns1="http://www.ech.ch/xmlns/eCH-0211/2" xmlns:ns2="http://www.ech.ch/xmlns/eCH-0097/2" xmlns:ns3="http://www.ech.ch/xmlns/eCH-0007/6">
  	<ns1:entryOfficeIdentification>
  		<ns2:uid>
  			<ns2:uidOrganisationIdCategorie>CHE</ns2:uidOrganisationIdCategorie>
  			<ns2:uidOrganisationId>123123123</ns2:uidOrganisationId>
  		</ns2:uid>
  		<ns2:localOrganisationId>
  			<ns2:organisationIdCategory>ebaube</ns2:organisationIdCategory>
  			<ns2:organisationId><!-- ORGANISATION_ID --></ns2:organisationId>
  		</ns2:localOrganisationId>
  		<ns2:organisationName>Leitbeh√∂rde Burgdorf</ns2:organisationName>
  		<ns2:legalForm>0223</ns2:legalForm>
  	</ns1:entryOfficeIdentification>
  	<ns1:municipality>
  		<ns3:municipalityName>Burgdorf</ns3:municipalityName>
  		<ns3:cantonAbbreviation>BE</ns3:cantonAbbreviation>
  	</ns1:municipality>
  </office>
  
  '''
# ---
